lib/src/rules/dartdoc_summary_line.dart

```diff
@@ -0,0 +1,181 @@
+void delete(String path) {
+  ...
+}
+```
+
+**BAD:**
+```dart
+/// Deletes the file at [path]. Throws an [IOError] if the file could not
+/// be found. Throws a [PermissionError] if the file is present but could
+/// not be deleted.
+void delete(String path) {
+  ...
+}
+```
+''';
+
+/// Enforces that DartDoc blocks start with a single, brief line that ends in a period.
+///
+/// (Like the one above!)
+class DartdocSummaryLine extends LintRule {
+  DartdocSummaryLine()
+      : super(
+            name: 'dartdoc_summary_line',
+            description: _desc,
+            details: _details,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(
+      NodeLintRegistry registry, LinterContext context) {
+    var visitor = _Visitor(this);
+    registry.addComment(this, visitor);
+  }
+}
+
+class _Visitor extends SimpleAstVisitor<void> {
+  final LintRule rule;
+
+  _Visitor(this.rule);
+
+  /**
+   * Visits a [Comment] node, and finds and validates its summary line.
+   */
+  @override
+  void visitComment(Comment node) {
+    if (!node.isDocumentation) {
+      // Skip things that may not be proper dartdoc (like a /** */ comment)
+      return;
+    }
+
+    List<String> summary = _getSummary(node).toList();
+
+    // Should we enforce that summary.length == 1?
+    if (!_endsWithPeriod(summary) ||
+        summary.join(' ').length > maxSummaryLength) {
+      // Highlight the offending comment
+      rule.reportLintForOffset(node.offset, 3);
+    }
+  }
+}
+
+// Some utility functions
+
+/**
+ * Detects if a Comment starts with slash-star-star.
+ *
+ * This is a less common style for Dart Docs, but still [valid](https://dart.dev/guides/language/language-tour#documentation-comments).
+ */
+bool _isSlashStarStar(Comment node) =>
+    node.tokens.first.lexeme.startsWith('/**');
+
+/// Removes the first appearance of `prefix` from `line`, and trims the output.
+String _trimPrefix(String line, String prefix) =>
+    line.replaceFirst(prefix, '').trim();
+
+/// Removes the first triple slash of a `line` and returns its trimmed value.
+String _trimSlashes(String line) => _trimPrefix(line, '///');
+
+/**
+ * Removes the first asterisk of a `line` and and returns its trimmed value.
+ *
+ * (Like the ones that make this comment!)
+ */
+String _trimStars(String line) => _trimPrefix(line, ' *');
+
+/// [String.isNotEmpty] so it can be torn off.
+bool _isNotEmpty(String line) => line.isNotEmpty;
+
+/// Retrieves the summary line of a [Comment] node.
+///
+/// This will take "lines" (tokens or actual strings) from the beginning of the
+/// node until an empty line is found.
+Iterable<String> _getSummary(Comment node) {
+  if (_isSlashStarStar(node)) {
+    // The first token contains the whole block comment... Split in lines, then
+    // take as many as needed...
+    LineSplitter splitter = LineSplitter();
+    List<String> lines = splitter.convert(node.tokens.first.lexeme);
+    return lines
+        .sublist(1, lines.length - 1)
```

Comment:
Looks like you need to do a range check first?

https://github.com/dart-lang/linter/actions/runs/3414458884/jobs/5682443674

Reply:
Yyyyep, I imagine this is caused by a "/** */" comment in a single line; I'm going to make less assumptions about those :P